### PR TITLE
Do not send fips_mode_enabled when it is not set (needed for vCD 9.0)

### PR DESCRIPTION
**Pull request by Didainius**
vCD 9.0 edge gateway does not support FIPS mode and API throws error if the XML field `FipsModeEnabled` is sent. This modifies edge gateway creation so that if the field `fips_mode_enabled` is not set - it is not sent.

The problem was identified here - https://github.com/terraform-providers/terraform-provider-vcd/issues/323#issuecomment-561575052

### CODE REVIEW

**Comment by Didainius** [comment]
> Looks good overall, only a few asks:
> 
>     1. If we have such, please reference the issue number or comment, where this was found in, in the description of the PR.
Added.
> 
>     2. Would it still be possible to share a test output which failed on 9.0 before the fix and succeeded after the fix?
Actually the same reference I added has the example output as well - https://github.com/terraform-providers/terraform-provider-vcd/issues/323#issuecomment-561575052

**Comment by Didainius** [comment]
> LGTM, only asking to confirm that `resource_vcd_edgegateway_test.go` (or other relevant test) is succeeding after this fix on your vCD 9.0 before merge too.

Sure- here is the output without the fix:
```shell
--- FAIL: TestAccVcdEdgeGatewayComplex (22.21s)
    testing.go:635: Step 0 error: errors during apply:

        Error: error creating edge gateway: error instantiating a new Edge Gateway: API Error: 400: [ 10acdb60-04fc-4287-a0bc-65d012ca63ae ] HTTP 400 Bad Request
         - cvc-complex-type.2.4.d: Invalid content was found starting with element 'FipsModeEnabled'. No child element is expected at this point.

          on /var/folders/h2/c60z324s6219c3hc_w28hbx80000gp/T/tf-test771078234/main.tf line 41:
          (source code not available)

```

Here is the output with fix:
```
=== RUN   TestAccVcdEdgeGatewayComplex
--- PASS: TestAccVcdEdgeGatewayComplex (108.14s)
```

However some other test `TestAccVcdEdgeGatewayExternalNetworks`has problems on vCD 9.0 which are not related to FIPS mode. Will link to a separate PR when it is raised...

File: vcd/resource_vcd_edgegateway.go [hunk 1]
```diff
@@ -349,6 +349,18 @@ func resourceVcdEdgeGatewayCreate(d *schema.ResourceData, meta interface{}) erro
 		},
 	}
 
+	// vCD 9.0 does not support FIPS Mode and fails if XML tag <FipsModeEnabled> is sent therefore
+	// field value must be sent only if user specified its value
+	if fipsModeEnabled, ok := d.GetOkExists("fips_mode_enabled"); ok {
+		if vcdClient.APIVCDMaxVersionIs("<= 29.0") { // vCD 9.0 or less
+			_, _ = fmt.Fprint(getTerraformStdout(), "WARNING! FIPS mode is only supported starting"+
```
Comment (vbauzysvmware) [inline vcd/resource_vcd_edgegateway.go]: what happens here if I set? - API error? maybe is worth it here send error and fail early?

Comment (Didainius) [reply vcd/resource_vcd_edgegateway.go]: I had a doubt about it. Here is how it looks - https://github.com/terraform-providers/terraform-provider-vcd/issues/323#issuecomment-562136931


Probably yes - I could fail early.

Comment (vbauzysvmware) [reply vcd/resource_vcd_edgegateway.go]: If you talking about `Invalid content was found starting with element 'FipsModeEnabled'. No child element is expected at this point.` Then my answer is that 100% we have fail with good explanation. @lvirbalas and @dataclouder can include they opinions too if it is needed

Comment (Didainius) [reply vcd/resource_vcd_edgegateway.go]: Changed

**Pull request merged**